### ISSUE TITLE
Entity Configuration custom IEqualityComparer

### ISSUE THREAD

**Issue by laitang2000** [jehugaleahsa/ChangeDetector#1]
Hi,

I'm trying to add a custom eqaulityComparer in the EntityConfiguration, but I can't seem to get it working. 

I get the error message "the expression must refer to a property"

`
public class DriverChangeDetector : EntityConfiguration<Driver>
    {
        public DriverChangeDetector()
        {
            Add(x => x.AccountNumber, "Account Number");
            Add(x => x.StartDate, "Start Date", new DateTimeEqualityComparer());
            Add(x => x.EndDate, "End Date", Formatters.FormatDateTime, new DateTimeEqualityComparer());

```
    }
}

public class DateTimeEqualityComparer : IEqualityComparer<DateTime?>
{
    public bool Equals(DateTime? x, DateTime? y)
    {
        var xUtc = x?.ToUniversalTime();
        var yUtc = y?.ToUniversalTime();

        if (DateTime.Equals(x, y))
        {
            return true;
        }
        else { return false;}
    }

    public int GetHashCode(DateTime? obj)
    {
        throw new NotImplementedException();
    }
}`
```

Can you point me in the right direction?

Cheers

**Comment by jehugaleahsa** [comment]
That error means the property you are configuring is:
1) Not a property (a field for example)
2) Not a direct property (x.Prop.Value)

I do have some tests testing nullable DateTimes.
https://github.com/jehugaleahsa/ChangeDetector/blob/master/ChangeDetector.Tests/ChangeDetectionTester.cs#L136

I played around with my unit tests to see if I could reproduce your issue
based on your example and didn't have any luck.

**Comment by laitang2000** [comment]
Awesome! my field was not a nullable date! 
Thank you.

This is a awesome little library btw!

**Comment by jehugaleahsa** [comment]
I modeled this library after Entity Framework. I've always wished they provided a more general-purpose change detector for entities. Even then, I wasn't always comparing database entities anyway. I made this in an application that did diffs for event logging, notified users how approvers modified their requests and tracked changes needed to update external systems. I've been able to use this library to take the Unit of Work pattern way beyond just database concerns. I'm glad your finding it useful. Feel free to share how your using.
